packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -387,64 +418,95 @@ export const splitFormatIntoSections = <TDate>(
-        return {
-          ...section,
-          separator: `\u2069${section.separator}\u2066`,
-          parsingSeparator: section.separator,
-        };
-      }
-      return section;
+  const cleanSections = sections.map((section, sectionIndex) => {
+    if (section.separator !== null && section.separator.includes(' ')) {
+      section.separator = `\u2069${section.separator}\u2066`;
+    } else if (section.separator === '/') {
+      section.separator = ' / ';
     }
-    return {
-      ...section,
-      separator: ' / ',
-      parsingSeparator: '/',
-    };
+
+    section.hasStartSeparator = sectionIndex === 0;
+
+    return section;
   });
+
+  return {
+    sections: cleanSections,
+    startSeparator,
+  };
 };
 
-export const createDateStrFromSections = (
+/**
+ * Some date libraries like `dayjs` don't support parsing from date with escaped characters.
+ * To make sure that the parsing works, we are building a format and a date without any separator.
+ */
+export const getDateFromDateSections = <TDate>(
+  utils: MuiPickersAdapter<TDate>,
   sections: FieldSection[],
-  willBeRenderedInInput: boolean,
+) => {
+  const formatWithoutSeparator = sections.map((section) => section.formatValue).join(' ');
+  const dateWithoutSeparatorStr = sections
+    .map((section) => getSectionVisibleValue(section, false))
+    .join(' ');
+
+  return utils.parse(dateWithoutSeparatorStr, formatWithoutSeparator);
+};
+
+export const createDateStrForInputFromSections = (
+  sections: FieldSection[],
+  startSeparator: string,
 ) => {
   const formattedArray = sections.map((section) => {
-    let sectionValueStr = getSectionVisibleValue(section, willBeRenderedInInput);
+    let sectionValueStr: string = '';
+    if (section.hasStartSeparator) {
+      sectionValueStr += startSeparator;
+    }
 
-    const separator = willBeRenderedInInput
-      ? section.separator
-      : section.parsingSeparator ?? section.separator;
+    sectionValueStr += getSectionVisibleValue(section, true);
 
-    if (separator != null) {
-      sectionValueStr += separator;
+    if (section.separator != null) {
+      sectionValueStr += section.separator;
     }
 
     return `${sectionValueStr}`;
   });
 
-  if (willBeRenderedInInput) {
-    // \u2066: start left-to-right isolation
-    // \u2067: start right-to-left isolation
-    // \u2068: start first strong character isolation
-    // \u2069: pop isolation
-    // wrap into an isolated group such that separators can split the string in smaller ones by adding \u2069\u2068
-    return `\u2066${formattedArray.join('')}\u2069`;
-  }
-  return formattedArray.join('');
+  // \u2066: start left-to-right isolation
+  // \u2067: start right-to-left isolation
+  // \u2068: start first strong character isolation
+  // \u2069: pop isolation
+  // wrap into an isolated group such that separators can split the string in smaller ones by adding \u2069\u2068
+  return `\u2066${formattedArray.join('')}\u2069`;
```

Comment:
If we only generate for input, those can be added in the `startSeparator`, and the last `separator`

Reply:
True
But for the single range input, I guess we would need to remove them from the last section of the `start` date and the first section of the `last` date, which could be more work and complexity.

Reply:
Good point